common/network-shuffle/src/main/java/org/apache/spark/network/shuffle/OneForOneBlockFetcher.java

```diff
@@ -58,29 +59,41 @@
   private final BlockFetchingListener listener;
   private final ChunkReceivedCallback chunkCallback;
   private TransportConf transportConf = null;
-  private File[] shuffleFiles = null;
+  private Boolean toDisk;
+  private Supplier<File> tmpFileCreater;
+  private Supplier<Boolean> shuffleBlockFetcherIteratorIsZombie;
 
   private StreamHandle streamHandle = null;
 
   public OneForOneBlockFetcher(
+    TransportClient client,
+    String appId,
+    String execId,
+    String[] blockIds,
+    BlockFetchingListener listener,
+    TransportConf transportConf) {
+    this(client, appId, execId, blockIds, listener, transportConf, false, null, null);
+  }
+
+  public OneForOneBlockFetcher(
       TransportClient client,
       String appId,
       String execId,
       String[] blockIds,
       BlockFetchingListener listener,
       TransportConf transportConf,
-      File[] shuffleFiles) {
+      Boolean toDisk,
+      Supplier<File> tmpFileCreater,
+      Supplier<Boolean> shuffleBlockFetcherIteratorIsZombie) {
     this.client = client;
     this.openMessage = new OpenBlocks(appId, execId, blockIds);
     this.blockIds = blockIds;
     this.listener = listener;
     this.chunkCallback = new ChunkCallback();
     this.transportConf = transportConf;
-    if (shuffleFiles != null) {
-      this.shuffleFiles = shuffleFiles;
-      assert this.shuffleFiles.length == blockIds.length:
-        "Number of shuffle files should equal to blocks";
-    }
+    this.toDisk = toDisk;
+    this.tmpFileCreater = tmpFileCreater;
+    this.shuffleBlockFetcherIteratorIsZombie = shuffleBlockFetcherIteratorIsZombie;
```

Comment:
shall we add an assert that, if `toDisk` is true, `tmpFileCreater` and `shuffleBlockFetcherIteratorIsZombie` can not be null. Or alternatively, we can drop the `toDisk` parameter, and use `toDisk = tmpFileCreater != null`